R/upload_publication.R

```diff
@@ -1,15 +1,19 @@
-#'@title upload a publication to sciencebase
-#'@param file a file path to the presentation for upload
-#'@param ... additional parameters passed to \code{\link[sbtools]{session_check_reauth}}
-#'@importFrom sbtools item_upload_create session_check_reauth current_session
-#'@export
-upload_publication <- function(file, ...){
+#' Upload a publication to sciencebase
+#'
+#' You must be logged into ScienceBase with \code{\link{sb_login}} to use this
+#' function
+#' 
+#' @param file a file path to the presentation for upload
+#' @importFrom sbtools item_upload_create current_session
+#' @importFrom mda.streams locate_folder
+#' @export
+upload_publication <- function(file){
   
-  session_check_reauth(...)
+  if(is.null(current_session())) stop("session is NULL. call sb_login() before uploading")
   
   pubs_folder <- locate_folder('publications')
   id_out <- item_upload_create(parent_id = pubs_folder, file)
-  url_out <- .url_from_id(id_out)
+  url_out <- paste0(sbtools:::pkg.env$url_item, id_out)
```

Comment:
yeah, i guess it's not, really. i'm just looking at the file list a lot and wanting that to be tidy to help me think.